src/builder.rs

```diff
@@ -1,55 +1,52 @@
-use crate::{Enr, EnrError, EnrKey, EnrPublicKey, Key, NodeId, MAX_ENR_SIZE};
-use bytes::{Bytes, BytesMut};
-use rlp::{Encodable, RlpStream};
+use crate::{update::Update, Enr, EnrError, EnrKey, NodeId};
+use bytes::Bytes;
+use rlp::Encodable;
 use std::{
     collections::BTreeMap,
     marker::PhantomData,
     net::{IpAddr, Ipv4Addr, Ipv6Addr},
 };
 
 /// The base builder for generating ENR records with arbitrary signing algorithms.
+#[derive(Clone)]
 pub struct EnrBuilder<K: EnrKey> {
-    /// The identity scheme used to build the ENR record.
-    id: String,
-
-    /// The starting sequence number for the ENR record.
-    seq: u64,
-
-    /// The key-value pairs for the ENR record.
-    /// Values are stored as RLP encoded bytes.
-    content: BTreeMap<Key, Bytes>,
-
-    /// Pins the generic key types.
-    phantom: PhantomData<K>,
+    enr: Enr<K>,
+    updates: Vec<Update>,
 }
 
 impl<K: EnrKey> EnrBuilder<K> {
     /// Constructs a minimal `EnrBuilder` providing only a sequence number.
     /// Currently only supports the id v4 scheme and therefore disallows creation of any other
     /// scheme.
     pub fn new(id: impl Into<String>) -> Self {
-        Self {
-            id: id.into(),
-            seq: 1,
-            content: BTreeMap::new(),
+        let id = rlp::encode(&id.into().as_bytes()).freeze();
+        // create a dummy Enr over which work is done
+        let enr = Enr {
+            seq: 0,
+            node_id: NodeId::new(&[0; 32]),
+            content: BTreeMap::from([(b"id".to_vec(), id)]),
+            signature: Vec::default(),
             phantom: PhantomData,
-        }
+        };
+        let updates = Vec::default();
+        Self { enr, updates }
     }
 
     /// Modifies the sequence number of the builder.
     pub fn seq(&mut self, seq: u64) -> &mut Self {
-        self.seq = seq;
+        self.enr.seq = seq;
```

Comment:
I think there is a subtlety here that may cause trouble. 

If I followed correctly, whatever value we set here, when the ENR does the update, the sequence number will increase. So in principle this will be whatever we set + 1. 
Also we can no longer create ENR's with 0 sequence numbers. 

Not sure if others are doing it or not, but maybe we do want users to set their own sequence numbers or at least modify them if they want to. 

If they want to re-use one they have made in the past, the ENR is still valid and they could do it anyway by making a whole new ENR, so I think we prob should support the functionality.